mkosi/__init__.py

```diff
@@ -5337,6 +5357,13 @@ def create_parser() -> ArgumentParserMkosi:
         action=BooleanAction,
         help="Generate split out root/verity/kernel images, too",
     )
+    group.add_argument(
+        "--reproducible",
+        metavar="BOOL",
+        action=BooleanAction,
+        default=False,
```

Comment:
Just to open that discussion: Is there an argument why the default should *not* be reproducible?

Reply:
I don't think we want by default zero out the mtime. The alternative would be to define `SOURCE_DATE_EPOCH` to the current build time, and set all files to this time. Then include the timestamp in the manifest.

Reply:
I think using `SOURCE_DATE_EPOCH` and
1. saving it to the manifest, and
2. passing it through to build/postinst/finalise script 

would make sense, but let's see what other's think about it.

Reply:
Hmm, then we need to think a little bit about the UX here.

* Should we demand a `manifest.json` when we use `--reproducible`? This is a chicken and egg problem.
* Error or warning if we pass `--reproducible` without setting `SOURCE_DATE_EPOCH` in the env?
  * `--reproducible` might not actually make a reproducible image by default. This could be confusing.
* Zeroing `mtime` could be a separate option.

We might need a new option section for reproducability then as well.

Reply:
I'm in agreement with @behrmann. In fact, I came here to make pretty much the exactly same comment. Making things reproducible by default is IMHO nicer and simpler. I.e. there should be no config option. If `SOURCE_DATE_EPOCH` is set, we use it. If not, we set it to current time. In both cases, we save the value to the manifest, so that the build can be reproduced later. Also, I think the variable should be passed to all scripts by default. I don't see any downside to doing that. They might use it or ignore it, but that's no worse than _status quo_.